runtime/lua/vim/lsp.lua

```diff
@@ -1299,7 +1300,12 @@ function lsp.start_client(config)
       -- The initial trace setting. If omitted trace is disabled ("off").
       -- trace = "off" | "messages" | "verbose";
       trace = valid_traces[config.trace] or 'off',
+      ---@param method string
     }
+    client.supports_dynamic_registration = function(method)
```

Comment:
Added this function so we can check if the client supports dynamic registration for a certain method.

I've also updated the register/unregister handlers to only log when an unsupported registration happens. The unsupported methods are also added to the log

Reply:
Since the initial client capabilities added when starting the client are not exposed anywhere, it's currently not possible to enable dynamicRegistration after the facts.

The registrations will be tracked, but `client.supports_method` will only check those that were enabled during start.

We could expose `client.client_capabilities`, but I would leave it as is for now

Reply:
I'd prefer inlining this into `supports_method` instead of exposing an additional method. Users could access `client.config.capabilities` if they need the information.

Reply:
done, added the function on `_dynamic` instead. Indeed not need to expose on client.